openghg/store/_eulerian_model.py

```diff
@@ -36,7 +36,10 @@ def __exit__(
         exc_val: Optional[BaseException],
         exc_tb: Optional[TracebackType],
     ) -> None:
-        self.save()
+        if exc_type is not None:
+            logger.exception(msg=f"{exc_type}, {exc_tb}")
```

Comment:
Two things:
1. I believe `logger.exception` is mainly called from within a try-except statement from looking at - https://docs.python.org/3/library/logging.html#logging.Logger.exception which says "This method should only be called from an exception handler.". So I think this should be [`logger.error()` instead](https://docs.python.org/3/library/logging.html#logging.Logger.error)
2. Did you want this to raise the error or just log it?

Reply:
Ohh OK, I must have got them mixed up. I'll change it to be logger.error now. I want the error to propagate.

Reply:
OK done